docs/vm-metrics.md

```diff
@@ -0,0 +1,178 @@
+To increase resilience, each node should expose all the metrics it knows about, avoiding
+a single point of failure per cluster. The resulting architecture is shown in the figure below.
+
+![VM metrics diagram](vm-metrics.png "VM Metrics")
+
+To have the full set the VM metrics, a guest agent like the QEMU Guest Agent should be run
+in the virtual machines. This step, albeit recommended, is totally left up to the administrator
+of the cluster.
+
+
+## VM metrics naming rules
+
+In order to minimize the clashes and make the VM metrics name predictable, we adopt the following rules
+to set the names:
+
+1. A name should be in the form `{VM name}.{group}.{metric name}`. The dot (`.`) is used as separator.
+   the `camelCase` should be used for composite words. Example: `VM0.migration.totalData`
+
+2. Abbreviations should be used only if their meaning is well known and not ambiguous. Examples:
+
+- **BAD**:
+  `VM0.migration.bps`
+  What `bps` is? Is it Bytes Per Second or Bits per Second? Or something else more specific?
+  I need to look it up in the documentation.
+
+3. If not obvious, the metric name should include the unit. Examples:
+
+- **BAD**: `VM0.migration.processedMemory`
+  What is it? bytes? KiBs? I need to look it up in the documentation.
+
+- **GOOD**: `VM0.migration.processedMemoryKiB`
+
+4. Keep the common prefixes among related metrics as long as possible. Examples
+
+- **BAD**: `VM0.migration.processedMemory`, `VM0.migration.remainingMemory`
+
+- **GOOD**: `VM0.migration.memoryProcessed`, `VM0.migration.memoryRemaining`
+
+Once name is exported, it is considered part of the API stability and treated accordingly.
+
+TODO: what about name clashes? at least report them? We can do that?
+
+
+## Consuming the VM metrics
+
+In order to make the task of the consumer easier, the VM metrics provider pod normalizes the
+exported dataset in the following manner
+
+- It always export all the metrics it knows about, even if the real data is not yet available.
+  Such metrics report the special value [TODO: which one? does this
+  concept exist in Kubernetes?] until the real data is available.
+  For example, the VM metric `VM.migration.dataTotal` will be always
+  reported even before the first migration was attempted, if ever.
+
+- The ephemeral metrics are kept even after libvirt no longer reports them, using the last
+  available data (e.g. migration downtime).
+
+However, the VM metrics provider pod will provide no guarantees about
+the durability of the data:
+
+- The consumer must take care to any form of persistence it may need.
+
+- The provider provide time series of data without holes in the sequence
+  (see normalization notes above).
+
+- The provider will never attempt to recover any past data.
+
+
+## Implementation notes
+
+To implement the VM metrics collection and reporting, we will initially
+adapt the [vAdvisor project](https://github.com/KubeVirt/vAdvisor),
```

Comment:
You're talking about the agent on the node collection?  Or something higher level?

Reply:
I mean the agent on the node, to report metrics we care about and cAdvisor doesn't know